### PR TITLE
KAA-1511: Update Java SDK documentation (for DesktopKaaPlatformContext class)

### PR DESCRIPTION
**Pull request by Serega290696**
…t class)

### CODE REVIEW

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 1]
```diff
@@ -155,7 +155,66 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+	 .setMaxLifeCycleThreads(CUSTOM_MAX_LIFE_CYCLE_THREADS) // set custom maximum life cycle threads amount
+	 .setMaxLifeCycleThreadsIdleMilliseconds(CUSTOM_MAX_LIFECYCLE_THREADS_IDLE_MILLISECONDS)
+	 // set custom maximum life cycle threads idle time
+	 .setMaxApiThreads(CUSTOM_MAX_API_THREADS) // set custom maximum api threads amount
+	 .setMaxApiThreadsIdleMilliseconds(CUSTOM_MAX_API_THREADS_IDLE_MILLISECONDS) // set custom maximum api threads idle time
+	 .setMaxCallbackThreads(CUSTOM_MAX_CALLBACK_THREADS) // set custom maximum callback threads amount
+	 .setMaxCallbackThreadsIdleMilliseconds(CUSTOM_MAX_CALLBACK_THREADS_IDLE_MILLISECONDS)//set custom maximum callback threads idle time
+	 .setMinScheduledThreads(CUSTOM_MIN_SCHEDULED_THREADS) // set custom minimum scheduled threads amount
+	 .build();
+new DesktopKaaPlatformContext(null, executorContext);
```
Comment (Rib47) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: Make **all your 3 examples** more comprehensive.
Every example must ends up with such code line:
`KaaClient client = Kaa.newClient( ... )`

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 2]
```diff
@@ -155,7 +155,66 @@ the boolean argument specifies whether to generate public/private key pair autom
...
- 
+
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of it's constructors. `DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. You can change it by calling corresponding constructor with specific `ExecutorContext` implementation.
+
+`FlexibleExecutorContext` has four thread groups, which responsible for life cycle, callbacks, API and scheduling. Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). Each of them has minimum and maximum threads amount and maximum thread idle time. By default minimum threads amount for all these groups is zero, maximum threads amount isn't limited and maximum idle time is 100 milliseconds. You can call corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). For scheduled thread groups, you can change only it's minimum threads amount. It is recommended to use builder to construct the `FlexibleExecutorContext` instance with custom parameters.
```
Comment (Rib47) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: Replace this:

> maximum threads amount isn't limited and maximum idle time is 100 milliseconds

With (defined more exactly):

> maximum threads amount **is almost not limited (it set to `Integer.MAX_VALUE` )** and maximum idle time is 100 milliseconds

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 3]
```diff
@@ -155,7 +155,66 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+
+The other ExecutorContext implementation which goes with Kaa is `SimpleExecutorContext`. Its minimum threads amount is 1 for each group. And its maximum threads amount also limited to 1 by default, but can be changed with proper constructor (see example below).
+
+Here some examples for using non-empty constructor and using builder.
+
+<ul class="nav nav-tabs">
+<li class="active"><a data-toggle="tab" href="#not-empty-constructor">DesktopKaaPlatformContext not empty constructor</a></li>
+<li><a data-toggle="tab" href="#builder">FlexibleExecutorContext builder</a></li>
+<li><a data-toggle="tab" href="#simpleExecutorContext">SimpleExecutorContext not empty constructor</a></li>
```
Comment (Rib47) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: Tab names is too long. Please remove "not empty constructor" part.

**Review by Rib47 — CHANGES_REQUESTED** [review]

**Comment by Rib47** [comment]
LGTM

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 4]
```diff
@@ -155,7 +155,71 @@ the boolean argument specifies whether to generate public/private key pair autom
 The first argument can be either `DesktopKaaPlatformContext` for Java desktop or `AndroidKaaPlatformContext` for Android applications. 
 In this example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
 And last one if set to **false** -- means using existing pre-generated public/private key pair, **true** -- means auto-generating new public/private key pair if absent.
+
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of it's constructors. `DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. You can change it by calling corresponding constructor with specific `ExecutorContext` implementation.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: Every sentence must start with a new line.

Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: s/it's constructors/its constructors

Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: s/calling corresponding/calling the corresponding

Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: s/with specific/with a specific

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 5]
```diff
@@ -155,7 +155,71 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+
+`FlexibleExecutorContext` has four thread groups, which responsible for life cycle, callbacks, API and scheduling. Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). Each of them has minimum and maximum threads amount and maximum thread idle time. By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. You can call corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). For scheduled thread groups, you can change only it's minimum threads amount. It is recommended to use builder to construct the `FlexibleExecutorContext` instance with custom parameters.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: s/which responsible for life cycle, callbacks, API and scheduling/responsible for life cycle, callbacks, API, and scheduling

**Review by akokhanovskyi — CHANGES_REQUESTED** [review]
Lots of grammar mistakes, I only marked a few in the beginning. Can you run the text through Grammarly?

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md (hunk 4 above)
Comment (akokhanovskyi) [reply doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: Also FYI, we have a documentation guide [here](http://kaaproject.github.io/kaa/docs/v0.10.0/Customization-guide/How-to-contribute/Documentation-contribution-and-style-guide/#markdown-formatting).

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 6]
```diff
@@ -155,7 +155,81 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. 
+You can call the corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). 
+For scheduled thread groups, you can change only it's minimum threads amount. 
+It is recommended to use the builder to construct the `FlexibleExecutorContext` instance with custom parameters.
+
+The other ExecutorContext implementation which goes with Kaa is `SimpleExecutorContext`. 
+Its minimum threads amount is 1 for each group. 
+And its maximum threads amount also limited to 1 by default but can be changed with proper constructor (see example below).
+
+Here some examples for using a non-empty constructor and using builder.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: are

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 7]
```diff
@@ -155,7 +155,81 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+
+`FlexibleExecutorContext` has four thread groups, responsible for life cycle, callbacks, API, and scheduling. 
+Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). 
+Each of them has minimum and maximum threads amount and maximum thread idle time. 
+By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. 
+You can call the corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). 
+For scheduled thread groups, you can change only it's minimum threads amount. 
+It is recommended to use the builder to construct the `FlexibleExecutorContext` instance with custom parameters.
+
+The other ExecutorContext implementation which goes with Kaa is `SimpleExecutorContext`.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: code

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 8]
```diff
@@ -155,7 +155,81 @@ the boolean argument specifies whether to generate public/private key pair autom
 The first argument can be either `DesktopKaaPlatformContext` for Java desktop or `AndroidKaaPlatformContext` for Android applications. 
 In this example, the second argument is the default implementation of `KaaClientStateListener` -- `SimpleKaaClientStateListener`, that is solely used to log the client state changes.
 And last one if set to **false** -- means using existing pre-generated public/private key pair, **true** -- means auto-generating new public/private key pair if absent.
+
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of its constructors. 
+`DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. 
+You can change it by calling corresponding the constructor with a specific `ExecutorContext` implementation.
+
+`FlexibleExecutorContext` has four thread groups, responsible for life cycle, callbacks, API, and scheduling. 
+Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass).
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: code

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 9]
```diff
@@ -155,7 +155,81 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+You can configure properties and `ExecutorContext` of `DesktopKaaPlatformContext` with help of its constructors. 
+`DesktopKaaPlatformContext` uses `FlexibleExecutorContext` implementation of `ExecutorContext` by default. 
+You can change it by calling corresponding the constructor with a specific `ExecutorContext` implementation.
+
+`FlexibleExecutorContext` has four thread groups, responsible for life cycle, callbacks, API, and scheduling. 
+Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). 
+Each of them has minimum and maximum threads amount and maximum thread idle time. 
+By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. 
+You can call the corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). 
+For scheduled thread groups, you can change only it's minimum threads amount.
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: its

File: doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md [hunk 10]
```diff
@@ -155,7 +155,81 @@ the boolean argument specifies whether to generate public/private key pair autom
...
+Each of these groups is implemented as separate thread pool (using ThreadPoolExecutor or its subclass). 
+Each of them has minimum and maximum threads amount and maximum thread idle time. 
+By default minimum threads amount for all these groups is zero, maximum threads amount is almost not limited (it set to `Integer.MAX_VALUE`) and maximum idle time is 100 milliseconds. 
+You can call the corresponding constructor to limit maximum threads amount or idle time (minimum threads amount is always zero). 
+For scheduled thread groups, you can change only it's minimum threads amount. 
+It is recommended to use the builder to construct the `FlexibleExecutorContext` instance with custom parameters.
+
+The other ExecutorContext implementation which goes with Kaa is `SimpleExecutorContext`. 
+Its minimum threads amount is 1 for each group. 
+And its maximum threads amount also limited to 1 by default but can be changed with proper constructor (see example below).
```
Comment (akokhanovskyi) [inline doc/Programming-guide/Using-Kaa-endpoint-SDKs/Java/index.md]: a proper constructor

**Review by akokhanovskyi — CHANGES_REQUESTED** [review]

**Pull request merged**